posthog/api/prompt.py

```diff
@@ -0,0 +1,100 @@
+from typing import Any, Dict, List, Union
+
+from dateutil import parser
+from rest_framework import exceptions, request, serializers, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.models.person.person import PersonDistinctId
+from posthog.models.prompt import PromptSequenceState, get_active_prompt_sequences
+
+
+class PromptSequenceStateSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = PromptSequenceState
+        fields = ["key", "last_updated_at", "step", "completed", "dismissed"]
+
+
+class PromptSequenceStateViewSet(StructuredViewSetMixin, viewsets.ViewSet):
+    """
+    Create, read, update and delete prompt sequences state for a person.
+    """
+
+    serializer_class = PromptSequenceStateSerializer
+
+    @action(methods=["PATCH"], detail=False)
+    def my_prompts(self, request: request.Request, **kwargs):
+        if not request.user.is_authenticated:  # for mypy otherwise unauthenticated users don't have distinct_id
```

Comment:
nit... should this be authenticated users in the comment (as opposed to unauthenticated)

Reply:
The double negation makes it confusing I guess, let me rephrase it

If we don't do a type check on authenticated users, mypy will throw an error because unauthenticated ones miss the distinct_id, which is required just after.